docs/metrics/extend/customresourcestate-metrics.md

```diff
@@ -292,6 +361,217 @@ spec:
 
 The above configuration was tested on [this](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/examples/hamster.yaml) VPA configuration, with an added annotation (`foo: 123`).
 
+#### All VerticalPodAutoscaler Metrics
+
+As an addition for the above configuration, here's the complete `CustomResourceStateMetrics` spec to re-enable all of the VPA metrics which are removed from the list of the default resources:
+
+<details>
+
+ <summary>VPA CustomResourceStateMetrics</summary>
+
+```yaml
+kind: CustomResourceStateMetrics
+spec:
+  resources:
+    - groupVersionKind:
+        group: autoscaling.k8s.io
+        kind: "VerticalPodAutoscaler"
+        version: "v1"
+      labelsFromPath:
+        namespace: [metadata, namespace]
+        target_api_version: [spec, targetRef, apiVersion]
+        target_kind: [spec, targetRef, kind]
+        target_name: [spec, targetRef, name]
+        verticalpodautoscaler: [metadata, name]
+      metricNamePrefix: "kube"
+      metrics:
+        # kube_verticalpodautoscaler_annotations
+        - name: "verticalpodautoscaler_annotations"
+          help: "Kubernetes annotations converted to Prometheus labels."
+          each:
+            type: Info
+            info:
+              labelsFromPath:
+                # annotation_*: [metadata, annotations]
+                name: [metadata, name]
+        # kube_verticalpodautoscaler_labels
+        - name: "verticalpodautoscaler_labels"
+          help: "Kubernetes labels converted to Prometheus labels."
+          each:
+            type: Info
+            info:
+              labelsFromPath:
+                # label_*: [metadata, labels]
+                name: [metadata, name]
+        # kube_verticalpodautoscaler_spec_updatepolicy_updatemode
+        - name: "verticalpodautoscaler_spec_updatepolicy_updatemode"
+          help: "Update mode of the VerticalPodAutoscaler."
+          each:
+            type: StateSet
+            stateSet:
+              labelName: "update_mode"
+              path: [spec, updatePolicy, updateMode]
+              list: ["Auto", "Initial", "Off", "Recreate"]
+        # Memory kube_verticalpodautoscaler_spec_resourcepolicy_container_policies_minallowed
+        - name: "verticalpodautoscaler_spec_resourcepolicy_container_policies_minallowed"
+          help: "Minimum memory resources the VerticalPodAutoscaler can set for containers matching the name."
```

Comment:
Try to make metric name shorter if it's possible

For example, vpa_policies_minallowed

Reply:
I'm trying to make it the same as the previous metrics which are removed, so I made it the same. If we changed the metrics name here, it defeats the purpose to re-expose the removed metrics. The plan is to have no parity between the metrics removed with this guide. wdyt @CatherineF-dev?

Reply:
Inferring the naming based on the [Prometheus conventions](https://prometheus.io/docs/practices/naming/), and as such, for instance, and taking into consideration Catherine's comment, this may be changed to:

```
vpa_spec_resourcepolicy_container_policies_minallowed_bytes
```

However, since the point is to polyfill these metrics, it makes sense to not break the previous names.